exchange/bidder_test.go

```diff
@@ -183,6 +183,47 @@ func TestRequestBidRemovesSensitiveHeaders(t *testing.T) {
 	assert.ElementsMatch(t, seatBid.httpCalls, expectedHttpCalls)
 }
 
+func TestSetGPCHeader(t *testing.T) {
+	server := httptest.NewServer(mockHandler(200, "getBody", "responseJson"))
+	defer server.Close()
+
+	requestHeaders := http.Header{}
+	requestHeaders.Add("Content-Type", "application/json")
+
+	bidderImpl := &goodSingleBidder{
+		httpRequest: &adapters.RequestData{
+			Method:  "POST",
+			Uri:     server.URL,
+			Body:    []byte("requestJson"),
+			Headers: requestHeaders,
+		},
+		bidResponse: &adapters.BidderResponse{
+			Bids: []*adapters.TypedBid{},
+		},
+	}
+
+	debugInfo := &config.DebugInfo{Allow: true}
+	ctx := context.Background()
+	ctx = context.WithValue(ctx, DebugContextKey, true)
+
+	bidder := adaptBidder(bidderImpl, server.Client(), &config.Configuration{}, &metricsConfig.DummyMetricsEngine{}, openrtb_ext.BidderAppnexus, debugInfo)
+	currencyConverter := currency.NewRateConverter(&http.Client{}, "", time.Duration(0))
+	seatBid, errs := bidder.requestBid(ctx, &openrtb2.BidRequest{}, "test", 1, currencyConverter.Rates(), &adapters.ExtraRequestInfo{GlobalPrivacyControlHeader: "1"}, true)
+
+	expectedHttpCall := []*openrtb_ext.ExtHttpCall{
+		{
+			Uri:            server.URL,
+			RequestBody:    "requestJson",
+			RequestHeaders: map[string][]string{"Content-Type": {"application/json"}, "Sec-Gpc": {"1"}},
+			ResponseBody:   "responseJson",
+			Status:         200,
+		},
+	}
+
+	assert.Empty(t, errs)
+	assert.ElementsMatch(t, seatBid.httpCalls, expectedHttpCall)
+}
```

Comment:
Thank you for adding a test. Could you please add one more test for when the `RequestData.Headers` is nil. I don't believe there is a requirement for bidders to set this value and since its underlying type is a map, it can be nil.

Reply:
I have added the test.

Reply:
Thank you. I had intended for the nil test to execute your new logic when the request headers are nil. Could you please rename the test `TestSetHeaderNil` to `TestSetGPCHeaderNil` and set `&adapters.ExtraRequestInfo{GlobalPrivacyControlHeader: "1"}` in the nil test.

Reply:
I'll get this done shortly. Thanks!

Reply:
I have updated the test and logic. This was something important that I missed. Thanks for pointing this out.